File: core/src/main/java/com/lomekwi/cave/ui/node/NumOutputRow.java
package com.lomekwi.cave.ui.node;

import com.kotcrab.vis.ui.widget.VisLabel;
import com.lomekwi.cave.pipeline.Node;
import com.lomekwi.cave.pipeline.num.NumFrame;

/**
 * NumFrame 输出端口显示：只读数值行。
 */
public final class NumOutputRow extends VisLabel implements PortRow {
    private final Node.OutPort<?> port;

    public NumOutputRow(Node.OutPort<?> port) {
        super(port.getName() + ": " + ((NumFrame) port.getData()).getVal());
        this.port = port;
    }

    @Override
    public Node.OutPort<?> getPort() {
        return port;
    }
}

File: core/src/main/scala/com/lomekwi/cave/ui/node/NumOutputRow.scala
package com.lomekwi.cave.ui.node

import com.kotcrab.vis.ui.widget.VisLabel
import com.lomekwi.cave.pipeline.Node
import com.lomekwi.cave.pipeline.num.NumFrame

/**
 * NumFrame 输出端口显示：只读数值行。
 */
final class NumOutputRow(port0: Node.OutPort[?]) extends VisLabel(port0.getName + ": " + port0.getData.asInstanceOf[NumFrame].getVal) with PortRow {
  private final val port: Node.OutPort[?] = port0

  override def getPort: Node.OutPort[?] = port
}
